### PR TITLE
add FindAietools

### PR DESCRIPTION
**Pull request by denolf**

### CODE REVIEW

**Review by jgmelber — APPROVED** [review]

File: FindAietools.cmake [hunk 1]
```diff
@@ -0,0 +1,216 @@
+###############################################################################
+#  Copyright (c) 2024, AMD, Inc.
+#  All rights recd cd cd served.
```
Comment (stephenneuendorffer) [inline FindAietools.cmake:3]: typo?

File: FindAietools.cmake [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+#
+## 3: Components
+# The following components are supported:
+###  3.1. AIE
+#  AIETOOLS_AIE_INCLUDE_DIR - AIE full include path
+#  AIETOOLS_AIE_LIBME - 'libme.a' with full path
+#  AIETOOLS_AIE_LIBC - 'libc.a' with full path
+#  AIETOOLS_AIE_LIBM - 'libm.a' with full path
+#  AIETOOLS_AIE_RUNTIME_INCLUDE_DIR - AIE runtime full include path
+#  AIETOOLS_AIE_LIBSOFTFLOAT - 'softfloat.a' with full path
```
Comment (stephenneuendorffer) [inline FindAietools.cmake:54]: Should this really just be a refactoring of FindVitis?  So FindVitis should call this?

File: unittests/findAietoolsTest/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+set(subDirLevels "../..")
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_CURRENT_SOURCE_DIR}/${subDirLevels})
+
+project(test_Find_Aietools)
+
+#find_package(Vitis COMPONENTS AIE)
+#find_package(Vitis COMPONENTS AIE2)
+#find_package(Vitis)
+#find_package(Vitis COMPONENTS AIE AIE2)
+find_package(Aietools 2022.2 COMPONENTS AIE AIE2)
```
Comment (stephenneuendorffer) [inline unittests/findAietoolsTest/CMakeLists.txt:16]: Not sure about the capitalizations here.  AIETools maybe?

File: FindAietools.cmake (hunk 1 above)
Comment (denolf) [reply FindAietools.cmake:3]: thanks for catching, yes.

File: FindAietools.cmake (hunk 2 above)
Comment (denolf) [reply FindAietools.cmake:54]: yes

Comment (denolf) [reply FindAietools.cmake]: FindVitis now calls FindAIETools in a backwards compatible way

File: FindAIETools.cmake [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+	list(APPEND AIETools_FIND_COMPONENTS "AIE")
+endif()
+
+#if Xilinx tools correctly installed they are added to $ENV{PATH} one of CMake's default search paths
+
+# Find aiecompiler to derive version
+find_program(AIETOOLS_AIECOMPILER aiecompiler)
+if(NOT AIETOOLS_AIECOMPILER)
+	message(STATUS "Unable to find aiecompiler")
+else(NOT AIETOOLS_AIECOMPILER)
```
Comment (fifield) [inline FindAIETools.cmake:81]: I'm guessing this `else` with a condition inside is a typo and doesn't do anything?
@denolf

File: FindAIETools.cmake [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+			set(AIETools_VERSION_MINOR ${aiecompilerVersionMinor})
+		endif()
+	endif()
+endif(NOT AIETOOLS_AIECOMPILER)
+
+# Find AIE tools
+find_program(AIETOOLS_XCHESSCC xchesscc)
+if(NOT AIETOOLS_XCHESSCC)
+	message(STATUS "Unable to find xchesscc")
+else(NOT VITIS_XCHESSCC)
```
Comment (fifield) [inline FindAIETools.cmake:122]: oh, there's more. what does this do?

**Pull request merged**